AuthenticationWorkflowTests.swift

```diff
@@ -0,0 +1,315 @@
+    
+    //MARK: Render Tests
+    func test_render_initial() {
+        let authenticationWorkFlow = AuthenticationWorkflow(authenticationService: AuthenticationService())
+        let expectedState = ExpectedState<AuthenticationWorkflow>(state:.emailPassword(nil))
+        
+        let expectedWorkflow = ExpectedWorkflow(
+            type: LoginWorkflow.self,
+            rendering: LoginScreen(
+                title: "",
+                email: "",
+                onEmailChanged: {_ in},
+                password: "",
+                onPasswordChanged: {_ in},
+                onLoginTapped: {}),
+            output: nil)
+        
+        let renderExpectations = RenderExpectations<AuthenticationWorkflow>(
+            expectedState: expectedState,
+            expectedOutput: nil,
+            expectedWorkers: [],
+            expectedWorkflows: [expectedWorkflow]
+        )
+        
+        
+        authenticationWorkFlow
+        .renderTester(initialState: .emailPassword(nil))
+        .render(
+            with: renderExpectations,
+            assertions: { screen in
+                
+            }
+        )
+        
+    }
+    
+    func test_render_AuthorizingEmailPasswordWorker() {
+        
+        let authenticationService = AuthenticationService()
+        let authenticationWorkFlow = AuthenticationWorkflow(authenticationService: authenticationService)
+
+        let expectedState = ExpectedState<AuthenticationWorkflow>(
+            state:.authorizingEmailPassword(
+                email: "reza@example.com",
+                password: "password"
+            )
+        )
+        
+        let expectedWorkflow = ExpectedWorkflow(
+            type: LoginWorkflow.self,
+            //NOTE: This rendering is not tested. why do we need it?
+            rendering: LoginScreen(
+                title: "",
+                email: "",
+                onEmailChanged: {_ in},
+                password: "",
+                onPasswordChanged: {_ in},
+                onLoginTapped: {}
+            ),
+            output: nil
+        )
+        
+        let expectedWorker = ExpectedWorker(
+            worker: AuthenticationWorkflow.AuthorizingEmailPasswordWorker(
+                authenticationService: authenticationService,
+                email: "reza@example.com",
+                password: "password"
+            )
+        )
+        
+        let renderExpectations = RenderExpectations<AuthenticationWorkflow>(
+            expectedState: expectedState,
+            expectedOutput: nil,
+            expectedWorkers: [expectedWorker],
+            expectedWorkflows: [expectedWorkflow]
+        )
+        
+        authenticationWorkFlow
+        .renderTester(
+            initialState: .authorizingEmailPassword(
+                email: "reza@example.com",
+                password: "password"
+            )
+        )
+        .render(
+            with: renderExpectations,
+            assertions: { screen in
```

Comment:
@asthatrivedi I'm not sure if this is related to the changes you've made but I could not call a screen that was of type `ModalContainerScreen<BackStackScreen>` in any of the render tests. A mere mention of a screen in the closure would result in a linker failure with the following:

ld: warning: Could not find or use auto-linked framework 'ModalContainer'
ld: warning: Could not find or use auto-linked framework 'BackStackContainer'
Undefined symbols for architecture x86_64:
  "type metadata for ModalContainer.ModalContainerScreenModal", referenced from:
      implicit closure #1 () throws -> Swift.Int in closure #4 (ModalContainer.ModalContainerScreen<BackStackContainer.BackStackScreen>) -> () in Development_Unit_TicTacToeTests.AuthenticationWorkflowTests.test_render_authorizingTwoFactorWorker() -> () in AuthenticationWorkflowTests.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

`Undefined symbol: type metadata for ModalContainer.ModalContainerScreenModal`